File: apis/DatabaseContextLibrary/Models/Shelter.cs
namespace DatabaseContextLibrary.models
{
    public class Shelter
    {
        public Guid Id { get; set; }
        public string UserName { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public Guid AddressId { get; set; }
        public string FullShelterName { get; set; }
        public bool IsAuthorized { get; set; } = false;
        public Address Address { get; set; }
    }
}

File: client/lib/shelter.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pet_share/address.dart';

part 'shelter.g.dart';

@JsonSerializable()
class User {
  User({
    required this.id,
    required this.userName,
    required this.phoneNumber,
    required this.email,
    required this.address,
  });

  String id;
  String userName;
  String phoneNumber;
  String email;
  Address address;

  Map<String, dynamic> toJson() => _$UserToJson(this);
  factory User.fromJson(Map<String, dynamic> json) => _$UserFromJson(json);
}

@JsonSerializable()
class Shelter extends User {
  Shelter({
    required super.id,
    required super.userName,
    required super.phoneNumber,
    required super.email,
    required super.address,
    required this.fullShelterName,
  });

  bool isAuthorized = false;
  String fullShelterName;

  @override
  Map<String, dynamic> toJson() => _$ShelterToJson(this);
  factory Shelter.fromJson(Map<String, dynamic> json) =>
      _$ShelterFromJson(json);
}
